src/action.ml

```diff
@@ -744,7 +745,8 @@ let rec exec t ~ectx ~dir ~env_extra ~stdout_to ~stderr_to =
         match mode with
         | Always -> files
         | If_corrected_file_exists ->
-          List.filter files ~f:(fun file -> Path.exists file.Promote.src)
+          List.filter files ~f:(fun { Promote. src; dst } ->
+            Path.exists src && Path.exists dst)
```

Comment:
Why do we need to make sure that the destination exists now?

Reply:
Basically we must not promote the file if it's generated. But maybe we should do that check in super_context.ml instead

Reply:
Thinking about this more, doing nothing when the source file doesn't exist seems wrong. For instance if the user generates a file containing some `[@@deriving_inline ...]` attribute, then we at least want to check that they are correct.

I guess `promote-if` could have the following behavior: print the diff but only promote if the source file exist as well.